crates/ruff/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -335,7 +335,9 @@ pub(crate) fn printf_string_formatting(
     }
 
     // Parse each string segment.
-    let mut format_strings = vec![];
+    let mut num_positional = 0;
+    let mut num_keyword = 0;
```

Comment:
Nit: A boolean for keywords seems sufficient
```suggestion
    let mut has_keyword_arguments = false;
```

Reply:
Ah, we need to know if there's _more than_ one positional argument. If we have exactly one positional argument, we can't fix it unfortunately :sob: